woocommerce/woocommerce-android
# Issue/4740 add support for incoming messages from sdk

Parent issue: #4661 
Closes: #4740 


### Description
This PR added support for displaying incoming messages from Stripe terminal SDK. Basically, it adds support for displaying messages coming from `onRequestReaderDisplayMessage` and `onRequestReaderInput`.

Stripe basically removed some methods and moved few methods into `BluetoothListener` in the v2 of SDK. [Here's some ](https://stripe.com/docs/terminal/sdk-migration-guide#connecting-to-a-reader)more info on that from Stripe's official doc

Please note that this PR implements functionality that existed in v1 of stripe SDK. But, the implementation logic is different because of the SDK changes.


### Testing instructions
Since the SDK just sends us a `REMOVE_CARD` message while collecting payment (This might be true only for the simulated reader?). Running unit tests and ensuring collecting payment flow works fine should be enough.

However, if you want to see the messages sent by the SDK in the UI. Then, we can add any of the messages in line no 280 for `REMOVE_CARD`. Add a delay of 2 seconds in `PaymentManager.kt` class inside `processPaymentIntent` method before each of the states. We need this delay since in the simulator the `collecting payment` state moves to `Processing state` as soon as we get the message from the SDK. It might be hard to notice the message in the UI without the delay. 


### Images/gif

You can see the incoming message (The message from the SDK is mapped to a different message on the UI for better understanding) at 0:33 seconds in the below video.

https://user-images.githubusercontent.com/1331230/133038976-509ac88c-d49c-479b-af58-8f094c12327c.mp4
